pkg/controller/tailoredprofile/tailoredprofile_controller.go

```diff
@@ -536,14 +536,30 @@ func (r *ReconcileTailoredProfile) getRulesFromSelections(tp *cmpv1alpha1.Tailor
 	}
 	return rules, nil
 }
-
 func (r *ReconcileTailoredProfile) getVariablesFromSelections(tp *cmpv1alpha1.TailoredProfile, pb *cmpv1alpha1.ProfileBundle) ([]*cmpv1alpha1.Variable, error) {
-	variableList := []*cmpv1alpha1.Variable{}
-	for _, setValues := range tp.Spec.SetValues {
+	// First pass: Build de-duplicated map of variables, keeping last occurrence when duplicates exist.
+	variables := make(map[string]string)
+
+	for i, setValue := range tp.Spec.SetValues {
+		if oldVal, seen := variables[setValue.Name]; seen {
+			// We found a duplicate. Warn that we will ignore the previous usage.
+			r.Eventf(tp, corev1.EventTypeWarning,
+				"DuplicatedSetValue",
+				"Variable '%s' appears multiple times in setValues. The operator will keep the last usage with value '%s' (position %d), ignoring the previous value '%s'. Specifying a variable multiple times using setValues will fail in a future release. Please remove duplicates from setValues as it introduces ambiguity.",
+				setValue.Name, setValue.Value, i, oldVal)
+		}
+		// Always use the last occurrence's value
```

Comment:
This is good to highlight and would make a good addition to the event message. Having the values in the message is great and helps users know which one is being used during the scan.

One minor improvement we could make would be to remove the "positioning" statement, since the positioning probably isn't as important as just knowing "hey, this is the last one I found, and it's the one I'm going to use".